docs/developer-guide/project-conventions.md

```diff
@@ -71,6 +71,11 @@ $ fab test --lint
 
 This will be checked by CI on every commit, so it's a good idea to run this locally before pushing your changes.
 
+We also have CI to check Django migrations, in order to help prevent any potential data issues. The CI will only run on a pull request, if any `/migrations` folders have been changed.
+
+While this won't entirely stop potential data issues, it will help to catch any potential issues by forcing the developer to check that their migrations are sound, before they
+are deployed.
+
```

Comment:
This is a tricky part for a review, I have a few **thoughts** - 

1) Check migrations
- How about using this in a sub section  `### Check migrations` - it could stand out within `Code style and standards`?
- Is it Ok to avoid using - we in this kind of documentation ex (_we also have CI_...)
- As a developer, I'd like to know 
     - _what triggers_ and 
     - _how to overcome them_ ?
As per your example https://github.com/nationalarchives/ds-wagtail/actions/runs/8048147808/job/21978871967 a Warning is issued and the CI stops.
- `### Check migrations`
 `Are affected on a pull request. When new migrations are added where some actions can cause potential loss of data. The CI stops with a Warning! unless its bypassed * ?` * - is it with some code changes?
- if its blocking the PR, then it is a Warning? or perhaps Error! or Alert!?


2) What should the developer do to continue when it stops? 
- Perhaps some mentions in `Local development gotchas` section ?

Reply:
Hi Allan, sorry for the long wait, I've updated the documentation changes you asked for here :)